airflow-core/src/airflow/api_fastapi/core_api/routes/public/extra_links.py

```diff
@@ -86,6 +86,22 @@ def get_extra_links(
         (link_name, task.get_extra_links(ti, link_name)) for link_name in task.extra_links
     )
     all_extra_links = {link_name: link_url or None for link_name, link_url in sorted(all_extra_link_pairs)}
+    # If no links found, or some are missing, try runtime (_link_*) keys from XCom
+    if not all_extra_links or any(link is None for link in all_extra_links.values()):
```

Comment:
I don't think this will work -- `task.extra` links will be based on the class attributes, so if a task defines _both_ at the class level and within the `__init__` level the init-based ones will be missed.